### ISSUE TITLE
SQL editor raises Multiple ResultSet if commented queries present (under some circumstances)

### ISSUE THREAD

**Issue by alexanderad** [metabase/metabase#2653]
If SQL editor has more than two queries defined and one is commented out `Multiple ResultSets were returned by the query.` is returned.

E.g.:

``` sql
select 1;
--select 2;
```

Result: `Multiple ResultSets were returned by the query.`.

**However** (for some reason):

``` sql
--select 1;
select 2;
```

Works as expected (returns 1 result).

See screencast:
![](http://i.imgur.com/ay81Ewo.gif)

We often use raw SQL editor while building queries and it is really troublesome to reorder commented queries so that all commented queries are on top. Please send help :)
-  I am using the Chrome browser.
-  My computer's OS is Linux.
-  I'm running Metabase locally from JAR.
-  My database is Postgres 9.5.
-  My Metabase version is 0.17.0.

**Comment by camsaul** [comment]
I'm guessing since there's text after the semicolon terminating your first query it tries to run it as a second, blank query. 

It might be possible to execute the queries in a different way where the presence of multiple result sets doesn't cause an exception to be thrown, but what if we have a query like this (nothing commented out)?

```sql
SELECT 1;
SELECT 2;
```

**Comment by alexanderad** [comment]
Perhaps I was not super clear with this dummy queries. This applies to any query (which has anything to return). It is being reproduced when commented out query is not last.

I just cleared out any white-spaces and newlines, and tried to work on 2 lines and 2 statements. Pay no attention to the return value, we don't really care for this bug what is being returned. Imagine I'm experimenting on a query that is why I'm commenting one of lines.

**Case 1:**
```sql
line 1: SELECT 1
line 2: SELECT 1
```
Result:  `Multiple ResultSets were returned by the query` -- **this is expected and correct behavior.**

**Case 2:**
```sql
line 1: --SELECT 1
line 2: SELECT 1
``` 
Result: `1` returned` -- this is OK (highlighter, btw does its job as well, first query is grayed out)

**Case 3:**
```sql
line 1: SELECT 1
line 2: --SELECT 1
```
Result: in this case "last" query is commented out but instead of returning `1`, metabase throws  `Multiple ResultSets were returned by the query`.

**Comment by alexanderad** [comment]
Working with raw queries often involves experiments before you finally get to what you want (a single query). This results of us sometimes having a workspace like:
```sql
-- have no distinct // select store_id, count(*) from stores group by npi ...
-- select store_id, count(dictinct npi) from stores group by ...
select count(*) from stores ...
-- too many // select store_id from stores
```
In this case after series of iterations I'm back to 3rd "variant", but in order to run it I have to reorder commented queries like:

```sql
--select store_id, count(*) from stores group by npi ...
-- select store_id, count(dictinct npi) from stores group by ...
--select store_id from stores
select count(*) from stores ...
```

During the work on a query metabase returns you result set if and only if your "current" query is last.

**Comment by xamguo** [comment]
Is this still a todo after more than three years?

**Comment by flamber** [comment]
This looks like it would be fixed by #8674 - if it gets merged.

**Comment by flamber** [comment]
This was fixed in 0.39.0 - likely by #14883
![image](https://user-images.githubusercontent.com/1447303/140615826-34fd8a0c-9f96-4d9b-b5de-9daa384971e4.png)
